### PR TITLE
Allow passing a shapefile as tagging_region input

### PR DESCRIPTION
**Pull request by BSchilperoort**
A user can now give a shapefile as tagging_region.

The mask will be created on-the-fly while tracking. The mask will also be written to output_folder/debug

### CODE REVIEW

File: src/wam2layers/config.py [hunk 1]
```diff
@@ -131,12 +131,18 @@ class Config(BaseModel):
...
+    a bounding box of the form [west, south, east, north].
+
+    The netcdf file should consist of ones (tagging region) and zeros 
+    (non-tagging region). 
+    Values between 0 and 1 are possible as well and can be used in case the
+    region of interest overlaps with part of a grid cell.
+
+    The shapefile should contain only one polygon. The mask generated from the
+    shapefile is also written to the debug directory in the output folder.
+    The entire shapefile polygon needs to be inside -180, -80, 180, 80.
```
Comment (ruudvdent) [inline src/wam2layers/config.py:144]: Let's put this into bullets

**Review by ruudvdent — CHANGES_REQUESTED** [review]
let's also update "[User guide](https://wam2layers.readthedocs.io/en/latest/userguide/index.html) Obtaining input data"

**Comment by BSchilperoort** [comment]
> let's also update "[User guide](https://wam2layers.readthedocs.io/en/latest/userguide/index.html) Obtaining input data"

Done!

File: docs/userguide/input.md [hunk 2]
```diff
@@ -13,10 +13,13 @@ Every tracking experiment requires data, which you might need to download.
 
 * **Tracking region**
```
Comment (ruudvdent) [inline docs/userguide/input.md:14]: 'Tagging' region

File: docs/userguide/input.md [hunk 3]
```diff
@@ -13,10 +13,13 @@ Every tracking experiment requires data, which you might need to download.
...
+```{eval-rst}
     In addition to the time-dependent forcing data, WAM2layers also needs a
-    netcdf-file containing a variable called `tagging_region`, with values between 0
-    and 1. It must have the same lat/lon coordinates as the other input data. This
-    file is used to define the tagging region from where to track moisture.
+    tagging region. This is used to define the tagging region from where to 
+    track moisture.
+    The region can be described by a netCDF file, shapefile, or simple
+    bounding box. For more info see :class:`wam2layers.config.Config.tagging_region`
+```
```
Comment (ruudvdent) [inline docs/userguide/input.md:22]: Alternatively, you may find other options for building your own tagging region under notebooks.

**Review by ruudvdent — APPROVED** [review]
approve subject to minor changes

**Pull request merged**